### PR TITLE
fix: Mark all equal preference as abstention for star and star-pr

### LINKED ISSUE #884: Update abstain behavior for STAR
Currently STAR ballots are marked as abstain if they're all null but this behavior should be changed to count as abstain if they're all equal aswell (all threes, all zeros, mix of zeros and nulls, etc)

There's 2 halves to this change

- [ ] **Frontend**: In the receipt after submitting a ballot, we need to change when it shows races as an abstain
- [ ] **Backend**: The tabulators use [makeAbstentionTest](https://github.com/Equal-Vote/star-server/blob/11df7afb1f079ea09916e86b436402171887cf1c/packages/backend/src/Tabulators/Util.ts#L105) to filter out ballots. I recommend adding another parameter for markAllEqualAsAbstention and have it default to false. Then the star.ts tabulator can pass in true for this

### PR DESCRIPTION
**Pull request by JonBlauvelt**
## Description
Mark all equal scores as abstention for star and star-pr

## Screenshots / Videos (frontend only) 

https://github.com/user-attachments/assets/9730b612-651d-4187-a7aa-fc0592606fdd

https://github.com/user-attachments/assets/b7539dae-3a1c-4a0b-bfce-a5f450781434

https://github.com/user-attachments/assets/3dfb88d9-63c4-49ab-8d59-7221533b809f

## Related Issues

fixes #884

### CODE REVIEW

**Comment by ArendPeter** [comment]
Looks like one of the tests is failing. It probably needs to be updated to account for the new abstention behavior

```
FAIL src/Tabulators/Star.test.ts (6.992 s)
  ● STAR Tests › Test valid/invalid/under/bullet vote counts

    expect(received).toBe(expected) // Object.is equality

    Expected: 6
    Received: 8

      1[29](https://github.com/Equal-Vote/bettervoting/actions/runs/15950520116/job/44989926631?pr=957#step:6:30) |
      130 |         const results = Star(...mapMethodInputs(candidates, votes), 1)
    > 1[31](https://github.com/Equal-Vote/bettervoting/actions/runs/15950520116/job/44989926631?pr=957#step:6:32) |         expect(results.summaryData.nTallyVotes).toBe(6);
          |                                                 ^
      1[32](https://github.com/Equal-Vote/bettervoting/actions/runs/15950520116/job/44989926631?pr=957#step:6:33) |         expect(results.summaryData.nOutOfBoundsVotes).toBe(2);
      1[33](https://github.com/Equal-Vote/bettervoting/actions/runs/15950520116/job/44989926631?pr=957#step:6:34) |         expect(results.summaryData.nAbstentions).toBe(2);
      1[34](https://github.com/Equal-Vote/bettervoting/actions/runs/15950520116/job/44989926631?pr=957#step:6:35) |     })

      at Object.<anonymous> (src/Tabulators/Star.test.ts:131:49)```

File: packages/backend/src/Tabulators/Star.test.ts [hunk 1]
```diff
@@ -115,11 +115,13 @@ describe("STAR Tests", () => {
     test("Test valid/invalid/under/bullet vote counts", () => {
         const candidates = ['Allison', 'Bill', 'Carmen']
         const votes = [
+            [5, 5, 5],
```
Comment (ArendPeter) [inline packages/backend/src/Tabulators/Star.test.ts:118]: oh, another edge case to consider is ``[0, null, null]`` . That should also be abstain since null counts as zero in STAR. I think adding back some ``?? 0`` expressions in the equality checks should do the trick

**Comment by ArendPeter** [comment]
Can you add .DS_Store to the .gitignore?

**Comment by ArendPeter** [comment]
> > Looks like one of the tests is failing. It probably needs to be updated to account for the new abstention behavior
> > ```
> > FAIL src/Tabulators/Star.test.ts (6.992 s)
> >   ● STAR Tests › Test valid/invalid/under/bullet vote counts
> > 
> >     expect(received).toBe(expected) // Object.is equality
> > 
> >     Expected: 6
> >     Received: 8
> > 
> >       1[29](https://github.com/Equal-Vote/bettervoting/actions/runs/15950520116/job/44989926631?pr=957#step:6:30) |
> >       130 |         const results = Star(...mapMethodInputs(candidates, votes), 1)
> >     > 1[31](https://github.com/Equal-Vote/bettervoting/actions/runs/15950520116/job/44989926631?pr=957#step:6:32) |         expect(results.summaryData.nTallyVotes).toBe(6);
> >           |                                                 ^
> >       1[32](https://github.com/Equal-Vote/bettervoting/actions/runs/15950520116/job/44989926631?pr=957#step:6:33) |         expect(results.summaryData.nOutOfBoundsVotes).toBe(2);
> >       1[33](https://github.com/Equal-Vote/bettervoting/actions/runs/15950520116/job/44989926631?pr=957#step:6:34) |         expect(results.summaryData.nAbstentions).toBe(2);
> >       1[34](https://github.com/Equal-Vote/bettervoting/actions/runs/15950520116/job/44989926631?pr=957#step:6:35) |     })
> > 
> >       at Object.<anonymous> (src/Tabulators/Star.test.ts:131:49)```
> > ```
> 
> That's interesting. It looks like this is a good place to add some test data for the new abstention behavior as well though I don't really understand 1. How this change broke the test or 2. How the test was passing before.

huh, that is weird. #2 makes sense to me, since the prior logic would count null as abstain, but #1 is also a mystery to me 🤔

**Review by ArendPeter — COMMENTED** [review]
Left some nitpicks

File: packages/backend/src/Tabulators/Star.test.ts (hunk 1 above)
Comment (JonBlauvelt) [reply packages/backend/src/Tabulators/Star.test.ts:118]: ah ok. thanks!

**Review by ArendPeter — APPROVED** [review]
Looks good!
